client/delegations.go

```diff
@@ -0,0 +1,213 @@
+	for i := 0; i < c.MaxDelegations; i++ {
+		d, ok := delegations.next()
+		if !ok {
+			return data.TargetFileMeta{}, ErrUnknownTarget{file, snapshot.Version}
+		}
+
+		// covers 5.6.{1,2,3,4,5,6}
+		verifier := verifiers[d.parent]
+		target, err := c.loadDelegatedTargets(snapshot, d.child.Name, verifier)
+		if err != nil {
+			return data.TargetFileMeta{}, err
+		}
+
+		// stop when the searched TargetFileMeta is found
+		if m, ok := target.Targets[file]; ok {
+			return m, nil
+		}
+
+		if target.Delegations != nil {
+			err := delegations.add(target.Delegations.Roles, d.child.Name)
+			if err != nil {
+				return data.TargetFileMeta{}, err
+			}
+
+			targetVerifier, err := verify.NewDelegationsVerifier(target.Delegations)
+			if err != nil {
+				return data.TargetFileMeta{}, err
+			}
+			verifiers[d.child.Name] = targetVerifier
+		}
+	}
+
+	return data.TargetFileMeta{}, ErrMaxDelegations{
+		File:            file,
+		MaxDelegations:  c.MaxDelegations,
+		SnapshotVersion: snapshot.Version,
+	}
+}
+
+func (c *Client) loadLocalSnapshot() (*data.Snapshot, error) {
+	if err := c.getLocalMeta(); err != nil {
+		return nil, err
+	}
+
+	rawS, ok := c.localMeta["snapshot.json"]
+	if !ok {
+		return nil, ErrNoLocalSnapshot
+	}
+
+	snapshot := &data.Snapshot{}
+	if err := c.db.Unmarshal(rawS, snapshot, "snapshot", c.snapshotVer); err != nil {
+		return nil, ErrDecodeFailed{"snapshot.json", err}
+	}
+	return snapshot, nil
+}
+
+// loadDelegatedTargets downloads, decodes, verifies and stores delegated targets
+func (c *Client) loadDelegatedTargets(snapshot *data.Snapshot, role string, verifier verify.DelegationsVerifier) (*data.Targets, error) {
+	var err error
+	fileName := role + ".json"
+	fileMeta, ok := snapshot.Meta[fileName]
+	if !ok {
+		return nil, ErrRoleNotInSnapshot{role, snapshot.Version}
+	}
+
+	// 5.6.1 download target if not in the local store
+	// 5.6.2 check against snapshot hash
+	raw, alreadyStored := c.localMetaFromSnapshot(fileName, fileMeta)
+	if !alreadyStored {
+		raw, err = c.downloadMetaFromSnapshot(fileName, fileMeta)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	target := &data.Targets{}
+	// 5.6.3 verify signature with parent public keys
+	// 5.6.5 verify that the targets is not expired
+	if err := verifier.Unmarshal(raw, target, role, fileMeta.Version); err != nil {
+		return nil, ErrDecodeFailed{fileName, err}
+	}
+	// 5.6.4 check against snapshot version
+	if target.Version != fileMeta.Version {
+		return nil, ErrTargetsSnapshotVersionMismatch{
+			Role:                     fileName,
+			DownloadedTargetsVersion: fileMeta.Version,
+			TargetsSnapshotVersion:   target.Version,
+			SnapshotVersion:          snapshot.Version,
+		}
+	}
+	// 5.6.6 persist
+	if !alreadyStored {
+		if err := c.local.SetMeta(fileName, raw); err != nil {
+			return nil, err
+		}
+	}
+	return target, nil
+}
+
+func (c *Client) rootTargetDelegation() data.DelegatedRole {
```

Comment:
Don't call it "rootTargetDelegation", please. Very confusing with "root" and "targets". Maybe "topLevelTargetsDelegations"? You get the idea.